### PR TITLE
Fix reading an empty trailing subpacket in v6 signatures

### PR DESCRIPTION
**Pull request by twiss**
Drive-by: add some tests for a key containing an unknown subkey and a trailing empty subpacket, both with and without `ignoreUnsupportedPackets: false`.

### CODE REVIEW

File: test/general/openpgp.js [hunk 1]
```diff
@@ -917,6 +917,101 @@ me8oJym/ILMHRGIc4JvY9+2wE5U1FBYTsze3WnVH5dP5mfA2Uk83TR5KewKANsb4kl/OEPlADWdR
 =c2mQ
 -----END PGP PRIVATE KEY BLOCK-----`;
 
+const experimentalPqcNistReadyPubKey = `-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (larabr) [inline test/general/openpgp.js:921]: Can you maybe write a comment about what the key structure actually is? Or where are these from? On the sequoia dumper I only see X25519/Ed25519 keys

Also, once NIST support is added I guess these tests will need updating? I think creating a key with fake algo IDs might be better?

Comment (twiss) [reply test/general/openpgp.js:921]: This key is from https://sequoia-pgp.gitlab.io/openpgp-interoperability-test-suite/pqc-nist-bp.html#Encrypt-Decrypt_roundtrip_with_v6_Ed25519_X25519_ML-KEM-768_ECDH-NIST-P-384_key and has a subkey with experimental algorithm ID 100, so it won't need updating unless we add support for that (which we shouldn't do in a production release).

**Review by larabr — APPROVED** [review]
